agent/agent.ts

```diff
@@ -0,0 +1,10 @@
+console.log('[agent] hello from agent');
+
+let testElement = document.getElementById('test-frame') as HTMLIFrameElement;
+
+testElement.src = "http://localhost:4001";
```

Comment:
This currently hard codes the address of the proxy server. I'm thinking that once this is all working, we'd receive this message via our connection to the connection server. The tricky question is actually how we'd set up *that* connection. 🤔

Reply:
```ts
import { Client } from '@bigtestl/client';

export function main*(client: Client) {
  let { applicationURL } = yield client.connect();

  let testElement = document.getElementById('test-frame') as HTMLIFrameElement;
  testElement.src = applicationURL;

  try {
    while (true) {
      let [message] = yield on(window, "message");
      // handle message.
    }
  } finally {
    testElement.removeAttribute('src');
  }